src/type/scalars.js

```diff
@@ -22,13 +22,14 @@ const MAX_INT = 2147483647;
 const MIN_INT = -2147483648;
 
 function serializeInt(value: mixed): number {
-  if (Array.isArray(value)) {
-    throw new TypeError(
-      `Int cannot represent an array value: ${inspect(value)}`,
-    );
+  let num = value;
```

Comment:
I feel like this logic could be simplified. It's really strange to me to have an early-return in an else block:
```
if (typeof value === 'boolean') {
  return value ? 1 : 0;
}

const num = (typeof value === 'string' && value !== '') ?
  Number(value) : 
  value;

if (!isInteger(num)) {
```